### DISCUSSION TITLE
Welcome to PyForecast Discussions!

### DISCUSSION THREAD

**Discussion by tjrocha** [DOI-BOR/PyForecast#98] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Nat-AlbertaRivers** [comment]
Hello!! 
I'm a river forecaster in Alberta, explporing the use of PyForecast as a method of updating the regression equations we use to estimate seasonal natural flows - it looks promising :)
Looks like we might need to write our own dataloader given that the stations we use are in Canada, and they are not part of the pre-loaded datasets. This is a bit over my head... but I'm workign towards figuring it out! Feel free to reach out if any of the work done to acquire data for stations in British Columbia could be applied to Alberta.
Cheers!

**Reply by tjrocha** [reply]
Hello. It would be fairly easy to do this if you have a web data service and a list of the sites and parameters you want to add. The files linked below along with the source code should point you in the right direction for how one might go about doing this. 

Alternatively, we can also build this in for you if we can show that we will mutually benefit from having your data included in the software. I work out of the US Bureau of Reclamation's Pacific Northwest region and we could use data from British Columbia in our forecasts. Does your data service have observations from sites in British Columbia?

@kevinfol - Would you folks be able to use Canadian data for river basins in your region?

Dataset Inventory: https://github.com/usbr/PyForecast/blob/NextFlow/Resources/GIS/PointDatasets.xlsx
Dataset Data Loaders: https://github.com/usbr/PyForecast/tree/NextFlow/Resources/DataLoaders

**Comment by beautah** [comment]
I recently inquired with Alberta's water group as to how I might get snow
pillow/course and precip data into our (NRCS) database - awdb. I never got
a reply but it's still on my to do list, currently our staff hand enter
data periodically but we'd love to have a data loader. My initial look into
webservices came up short but I did notice some period of record csvs (or
the like) could be obtained by some url parsing. If you can have whoever
handles your data delivery reach out to me at ***@***.*** I'd love
to hear about how we might be able to benefit each other. Once your data is
in AWDB you could simply use our data loader, kill two birds with one
stone. You can see the sites we currently hand enter data for here:

https://www.nrcs.usda.gov/wps/portal/wcc/home/quicklinks/imap#version=158&elements=&networks=!&states=AB&counties=!&hucs=&minElevation=&maxElevation=&elementSelectType=all&activeOnly=true&activeForecastPointsOnly=false&hucLabels=false&hucIdLabels=false&hucParameterLabels=true&stationLabels=&overlays=&hucOverlays=state&basinOpacity=100&basinNoDataOpacity=100&basemapOpacity=100&maskOpacity=0&mode=stations&openSections=elements,location,networks&controlsOpen=true&popup=&popupMulti=&popupBasin=&base=esriNgwm&displayType=inventory&basinType=6&dataElement=WTEQ&depth=-8&parameter=PCTMED&frequency=DAILY&duration=I&customDuration=&dayPart=E&monthPart=E&forecastPubDay=1&forecastExceedance=50&seqColor=1&divColor=3&scaleType=D&scaleMin=&scaleMax=&referencePeriodType=POR&referenceBegin=1981&referenceEnd=2010&minimumYears=20&hucAssociations=true&relativeDate=-1&lat=51.031&lon=-113.797&zoom=6.5

ideally we'd like to import any data you have, let me know if you have
someone I can talk to about pursuing this

Beau Uriona

**Comment by tjrocha** [comment]
Yes, this would be a good option @Nat-AlbertaRivers. We already have NRCS' data loader built in and it would be super easy to get your data incorporated into the software this way.

**Comment by Nat-AlbertaRivers** [comment]
Great! I can put you in touch with our data stewardship team to access the data, the snow pillows should be easy. We also have snow courses and precip that might be useful to you. We can talk offline about that, and perhaps one of you could give us a quick demo - the version I have doesn't quite match the wiki example. 
@beautah, I can't see your contact info above, could you send me a message at n a t h a l i e . b r u n e t _ at _ gov . ab . ca
Thanks!!!

**Reply by tjrocha** [reply]
@Nat-AlbertaRivers  - @kevinfol and I would be more than happy to give a demo and walkthrough. Let me coordinate with a few folks on our end to make sure we're not violating any inter-governmental stuff and we'll reach out to coordinate. Thank you for your interest in PyForecast!

-- Jon

**Reply by Nat-AlbertaRivers** [reply]
Sounds good! :)
